### PR TITLE
6 get api articles articleid comments

### PR DESCRIPTION
**Pull request by ZeusLGR**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -137,3 +137,70 @@ describe("GET /api/articles/:article_id", () => {
...
+        expect(comments).toBeInstanceOf(Array);
+        expect(comments).toHaveLength(2);
+        comments.forEach((comment) => {
+          expect(comment).toEqual(
+            expect.objectContaining({
+              comment_id: expect.any(Number),
+              votes: expect.any(Number),
+              created_at: expect.any(String),
+              author: expect.any(String),
+              body: expect.any(String),
```
Comment (popatre) [inline __tests__/app.test.js:144]: You would need to also check that these comments have the article_id that you've requested

File: __tests__/app.test.js [hunk 2]
```diff
@@ -137,3 +137,70 @@ describe("GET /api/articles/:article_id", () => {
...
+      });
+  });
+  test("status:200, responds with an empty array if the article_id exists in the db but there are no comments related to it", () => {
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+        expect(comments).toBeInstanceOf(Array);
+        expect(comments).toHaveLength(0);
```
Comment (popatre) [inline __tests__/app.test.js:166]: Nice one

File: __tests__/app.test.js [hunk 3]
```diff
@@ -137,3 +137,70 @@ describe("GET /api/articles/:article_id", () => {
...
+      });
+  });
+  test("status:404, responds with an error message if the route does not exist", () => {
+    return request(app)
+      .get("/api/notARoute/1/comments")
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Route not found");
+      });
+  });
```
Comment (popatre) [inline __tests__/app.test.js]: I'd suggest you don't need this, as i presume you set up the handler right at the start

Comment (popatre) [reply __tests__/app.test.js]: You probably already have a test for a bad route somewhere

File: controllers/controllers.js [hunk 4]
```diff
@@ -39,3 +41,18 @@ exports.getArticleByID = (req, res, next) => {
...
+    selectCommentsByArticleID(article_id),
+    checkIfArticleIDExists(article_id),
+  ];
+  Promise.all(promises)
+    .then(([comments]) => {
+      return res.status(200).send({ comments });
+    })
+    .catch((err) => {
+      next(err);
+    });
```
Comment (popatre) [inline controllers/controllers.js:57]: Good stuff

File: models/models.js [hunk 5]
```diff
@@ -42,3 +42,38 @@ exports.selectArticleByID = (article_id) => {
...
+
+exports.selectCommentsByArticleID = (article_id) => {
+  let SQL = `
+    SELECT comments.comment_id, comments.votes, comments.created_at, comments.author, comments.body
+    FROM comments
+    JOIN articles
+    ON comments.article_id = articles.article_id
+    WHERE comments.article_id = $1
+    GROUP BY comments.comment_id
+    ORDER BY comments.created_at
```
Comment (popatre) [inline models/models.js]: Im pretty sure the comments have an article id that you can just search for, im not sure what the JOIN is doing.

**Review by popatre — CHANGES_REQUESTED** [review]

File: __tests__/app.test.js (hunk 3 above)
Comment (ZeusLGR) [reply __tests__/app.test.js]: Shoud I separate out the bad route test into its own describe block and apply to all? At the moment I've been writing it for each new request

File: __tests__/app.test.js (hunk 1 above)
Comment (ZeusLGR) [reply __tests__/app.test.js:144]: Should I add article_id into the expect.objectContaining block? 

I wasn't sure if the returned comment objects have to exclusively include the properties stated on the trello card

File: models/models.js [hunk 6]
```diff
@@ -42,3 +42,35 @@ exports.selectArticleByID = (article_id) => {
     return article;
   });
 };
+
+exports.selectCommentsByArticleID = (article_id) => {
+  let SQL = `
+    SELECT *
+    FROM comments
```
Comment (popatre) [inline models/models.js:49]: Much better this

**Review by popatre — APPROVED** [review]
Looks good
In regards to the bad path - once you've tested one bad path, you've tested them all. 
You dont need to check again for every variation of a bad path.

 Your handler will catch any path that isnt listed in the app.js file

**Pull request merged**